### ISSUE TITLE
Difference in rol left result

### ISSUE THREAD

**Issue by developerandre** [flutter/flutter#15924]
Hi,
The result of the operator shift left << and right >> differs from the result in javascript.
 How to solve that.
Example: 
  var num = 123,cnt = 30;
  (num << cnt) | (num >>> (32 - cnt)); // in Dart 3221225502 and in JS  -1073741794

**Comment by zoechi** [comment]
Numbers in JS are always double and interger values are limited to 2^53 
while in Dart 2 int is 2^64

Making them behave equally would have too much of a performance impact.
The docs don't seem to be updated to Dart 2 yet

Is there a reason why you posted this in Flutter?

**Comment by developerandre** [comment]
Thanks @zoechi 
Yes I am creating a package for xmpp in flutter. And I need to implement that for the authentication mechanisms.
Currently the result is different from what I get in JS and the xmpp server returns me "username or password incorrect" in dart while it works in JS.
Note that I rely on stropheJs

**Comment by zoechi** [comment]
What are yiu using in Dart (for Flutter? strooheJs is only available in the browser

**Comment by developerandre** [comment]
Yes it is only for browsers but I removed the browser dependencies in the source code of stropheJs to adapt it to flutter

**Comment by zoechi** [comment]
How is JavaScript related then?
